diff --git a/src/uv.sw.js b/src/uv.sw.js
index e3dbf40..4057bd2 100644
--- a/src/uv.sw.js
+++ b/src/uv.sw.js
@@ -122,7 +122,7 @@ class UVServiceWorker extends Ultraviolet.EventEmitter {
 				? "blob:" + location.origin + requestCtx.url.pathname
 				: requestCtx.url;
 
 			const response = await this.bareClient.fetch(fetchedURL, {
-				headers: requestCtx.headers,
+				headers: Object.entries(requestCtx.headers),
 				method: requestCtx.method,
 				body: requestCtx.body,
 				credentials: requestCtx.credentials,
